activerecord/lib/active_record/connection_adapters/abstract/connection_pool.rb

```diff
@@ -803,7 +803,12 @@ def checkout_new_connection
 
         def checkout_and_verify(c)
           c._run_checkout_callbacks do
-            c.verify!
+            ActiveSupport::Notifications.instrumenter.instrument(
```

Comment:
Certainly. I had considered that but decided to use the instrumentation here to match what was already being used for this feature. Is there any significant difference between callbacks vs. notifications, or is it more of an arbitrary choice, and here just a matter of not needing to use both at the same time?